src/Node/DelimitedList.php

```diff
@@ -31,4 +31,17 @@ public function addToken($node) {
         }
         $this->children[] = $node;
     }
+
+    /**
+     * @return Node[]
+     */
+    public function getListElements() {
+        $listElements = [];
+        foreach ($this->children as $child) {
+            if($child instanceof Node) {
+                $listElements[] = $child;
```

Comment:
> We already have DelimiterLists that do not have a constant delimiter and there may be more.

We do? Sounds like a bug :smiley:. Could you elaborate?

In general, we keep away from holding any "descriptor" properties on Nodes because the memory cost of additional properties is 12-16 bytes each (depending on 32 bit or 64 bit), which very quickly adds up on large code bases.